xarray/plot/utils.py

```diff
@@ -221,8 +221,14 @@ def _infer_xy_labels(darray, x, y):
         if darray.ndim != 2:
             raise ValueError('DataArray must be 2d')
         y, x = darray.dims
-    elif x is None or y is None:
-        raise ValueError('cannot supply only one of x and y')
+    elif x is None:
+        if y not in darray.dims:
+            raise ValueError('y must be a coordinate variable')
```

Comment:
See the following example:

```
import xarray as xr
import numpy as np
lon, lat = np.meshgrid(np.linspace(-20, 20, 5), np.linspace(0, 30, 4))
da = xr.DataArray(np.arange(20).reshape(4, 5), dims=['y', 'x'],
                  coords = {'lat': (('y', 'x'), lat),
                            'lon': (('y', 'x'), lon)})
da.plot(x='lat')
```

Note that ``da.plot(x='lat', y='lon')`` is a valid call and that ``'lat' in da.coords`` is ``True`` (but ``'lat' in da.dims`` is ``False``).

With your current PR the code  snippet above would lead to the following error:

```x must be a coordinate variable```

Which is misleading, since ``'lat'`` *is* a coordinate variable (it is not a dimension coordinate, however). 

I just opened an issue to discuss these terminology problems: https://github.com/pydata/xarray/issues/1295